app/src/main/java/com/gatchimath/totask/Task.java

```diff
@@ -0,0 +1,108 @@
+package com.gatchimath.totask;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Basic task class.
+ *
+ * @author  gatchi (github.com/gatchi)
+ */
+class Task {
+	
+	private String name;
+	private String text;
+	private int id;
+	private boolean doneState;
+	
+	Task() {
+		this("");
+	}
+	
+	Task(String name) {
+		this(name, "");
+	}
+	
+	Task(String name, String text) {
+		this.name = name;
+		this.text = text;
+		this.id = 0;
+		this.doneState = false;
+	}
+	
+	Task(String name, String text, int id) {
+		this.name = name;
+		this.text = text;
+		this.id = id;
+		this.doneState = false;
+	}
+	
+	/**
+	 * @return The name and main content of the task.
+	 */
+	String getName() {
+		return name;
+	}
+	
+	/**
+	 * @param name  The name and main content of this task.
+	 */
+	void setName(String name) {
+		this.name = name;
+	}
+	
+	/**
+	 * @return Any additional details pertinent to the task.
+	 */
+	String getText() {
+		return text;
+	}
+	
+	/**
+	 * @param text  Any additional details pertinent to the task.
+	 */
+	void setText(String text) {
+		this.text = text;
+	}
+	
+	/**
+	 * @return True if the task is done, false if not.
+	 */
+	boolean isDone() {
+		return doneState;
+	}
+	
+	/**
+	 * Mark the task as done.
+	 */
+	void setDone() {
+		doneState = true;
```

Comment:
Implement these in terms of setDoneState, and make them final, as they are fully expressible in terms of that function. That way, in a derived class, only setDoneState need be (and indeed, can be) overridden.